eclipse-che/che-theia
# fix: Adapt the Secrets plugin API to use kubernetes secrets

Signed-off-by: Igor Vinokur <ivinokur@redhat.com>



### What does this PR do?
Rework the Secrets API from upstream theia to use kubernetes secrets

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

fixes https://github.com/eclipse/che/issues/19837

### How to test this PR?

1. Start a workspace from devfile:
```
apiVersion: 1.0.0
metadata:
  name: wksp-xhxvu
components:
  - type: cheEditor
    reference: 'https://raw.githubusercontent.com/chepullreq4/pr-check-files/master/che-theia/pr-1166/che_theia_meta.yaml'
  - type: chePlugin
    reference: 'https://pastebin.com/raw/riqiReMi'
```
2. Run Test Get Password command and see a notification with empty (undefined) password.
3. Run Test Set Password command and see a notification about password change.
4. Run Test Get Password command and see a notification with the updated password.
5. Run Test Delete Password command and see a notification about password change.
6. Run Test Get Password command and see a notification with empty (undefined) password.

P.S. This will work only for newly created namespace, see https://github.com/eclipse-che/che-server/pull/44#pullrequestreview-699942442
### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable
